### ISSUE TITLE
Request for Code for AutoAttack on AUC

### ISSUE THREAD

**Issue by jkcjkc** [AdaptiveMotorControlLab/AROS#17]
Hi! Thanks a lot for your inspiring work! I would like to request access to the specific code for the AutoAttack method on AUC. Could you kindly share it with me or provide any guidance on how to access it? I would greatly appreciate your help and any additional suggestions or resources you might have regarding this implementation.

**Comment by MMathisLab** [comment]
Hey @jkcjkc thanks so much for your interest! We will push our code asap for this 💜 cc @hsirm

**Comment by hsirm** [comment]
@jkcjkc  Thank you for your interest in our work.

We currently use the [**torchattacks**](https://github.com/Harry24k/adversarial-attacks-pytorch/tree/master/torchattacks/attacks) library for adversarial attacks, specifically **PGD (L2)** and **PGD (L∞)** in our code. We will be adding more evaluations soon, but in the meantime, you can easily test other attacks by defining them yourself and passing them into: ```aros_node.evaluate.get_auc_adversarial```. Regarding AutoAttack—which ensembles multiple attacks such as APGD, APGDT, FAB, and Square—we suggest dropping the DLR loss for APGD. This is because DLR requires more than three classes, which is not the case for OOD detection, where we essentially treat the problem as a binary classification (ID vs. OOD). Instead, you can replace DLR with another attack like PGD (as you can find in the appendix).

Please note that attacks is originally implemented for standard classification tasks. To adapt it for AROS, you need to adjust the objective function to target the OOD score rather than classification loss. Below is an example from our ```PGD_AUC``` function:
```
ones = torch.ones_like(labels)
multipliers = +1 * (ones - 2 * ones * (labels == self.num_classes))

for _ in range(self.steps):
    adv_images.requires_grad = True
    outputs = self.get_logits(adv_images)
    final_outputs = softmax(outputs)
    OOD_score = -torch.max(final_outputs, dim=1)[0]
    cost = torch.sum(OOD_score * multipliers)```



Here, we compute and targeted the OOD score (rather than a standard classification loss i.e. CE).

**Comment by jkcjkc** [comment]
Thank you for your patient and detailed response, and we look forward to more of your contributions to the community.
